AlphaWallet/Transactions/ViewModels/TransactionDetailsViewModel.swift

```diff
@@ -4,164 +4,190 @@ import BigInt
 import Foundation
 import UIKit
 import AlphaWalletFoundation
+import Combine
 
-struct TransactionDetailsViewModel {
-    private let transactionViewModel: TransactionViewModel
+struct TransactionDetailsViewModelInput {
+    let openUrl: AnyPublisher<Void, Never>
+    let copyToClipboard: AnyPublisher<TransactionDetailsViewModel.CopyableField, Never>
+}
+
+struct TransactionDetailsViewModelOutput {
+    let viewState: AnyPublisher<TransactionDetailsViewModel.ViewState, Never>
+    let copiedToClipboard: AnyPublisher<String, Never>
+    let openUrl: AnyPublisher<URL, Never>
+}
+
+class TransactionDetailsViewModel {
     private let transactionRow: TransactionRow
     private let chainState: ChainState
     private let fullFormatter = EtherNumberFormatter.full
-    private let coinTicker: CoinTicker?
-
-    private var server: RPCServer {
-        return transactionRow.server
+    private let analytics: AnalyticsLogger
+    private let transactionsService: TransactionsService
+    private let tokensService: TokenViewModelState
+    private let wallet: Wallet
+    private var moreButtonTitle: String {
+        if let name = ConfigExplorer(server: server).transactionUrl(for: transactionRow.id)?.name {
+            return R.string.localizable.viewIn(name)
+        } else {
+            return R.string.localizable.moreDetails()
+        }
     }
+    private var detailsURL: URL? { ConfigExplorer(server: server).transactionUrl(for: transactionRow.id)?.url }
+
+    var shareItem: URL? { detailsURL }
+    var server: RPCServer { transactionRow.server }
+    var shareAvailable: Bool { detailsAvailable }
 
-    init(
-            transactionRow: TransactionRow,
-            chainState: ChainState,
-            wallet: Wallet,
-            coinTicker: CoinTicker?
-    ) {
+    init(transactionsService: TransactionsService, transactionRow: TransactionRow, chainState: ChainState, wallet: Wallet, tokensService: TokenViewModelState, analytics: AnalyticsLogger) {
+        self.wallet = wallet
+        self.tokensService = tokensService
+        self.transactionsService = transactionsService
         self.transactionRow = transactionRow
         self.chainState = chainState
-        self.coinTicker = coinTicker
-        self.transactionViewModel = TransactionViewModel(
-            transactionRow: transactionRow,
-            chainState: chainState,
-            wallet: wallet
-        )
-    }
-
-    var title: String {
-        return R.string.localizable.transactionNavigationTitle()
-    }
+        self.analytics = analytics
+    }
+
+    func transform(input: TransactionDetailsViewModelInput) -> TransactionDetailsViewModelOutput {
+        let transactionRow = transactionsService.transactionPublisher(for: transactionRow.id, server: transactionRow.server)
+            .compactMap { $0 }
+            .receive(on: RunLoop.main)
+            .map { transaction -> TransactionRow in
+                if transaction.localizedOperations.count > 1 {
```

Comment:
@hboon do you think this can brake a screen as transaction row can have `.item(transaction, operation)` type.  so out of here we can't figure out right transaction row for transaction. TransactionRow builds based on localized operations, and thay can be changed when transaction has updated

Reply:
Not exactly sure of the scenario, but a transaction should only change state when it's pending -> confirmed, so they should be at most a handful of pending transactions and they should all be at the top, sorted by nonce. Does that help? If not, can you describe more?

Reply:
so in best case transaction row will stay the same, that willn't cause issues